src/main/java/uk/sky/cirrus/ClusterHealth.java

```diff
@@ -0,0 +1,35 @@
+package uk.sky.cirrus;
+
+import com.datastax.driver.core.Cluster;
+import com.datastax.driver.core.Host;
+import uk.sky.cirrus.exception.ClusterUnhealthyException;
+
+import java.net.InetAddress;
+import java.util.List;
+import java.util.stream.Collectors;
+
+class ClusterHealth {
+
+    private final Cluster cluster;
+
+    ClusterHealth(Cluster cluster) {
+        this.cluster = cluster;
+    }
+
+    void check() throws ClusterUnhealthyException{
+
+        final List<InetAddress> unhealthyHosts = cluster.getMetadata().getAllHosts()
+                .parallelStream()
```

Comment:
Unless we're expecting cassandra clusters with millions of nodes sequential will be faster.